### DISCUSSION TITLE
Unclear Source Code from a story

### DISCUSSION THREAD

**Discussion by seahindeniz** [storybookjs/storybook#27101] [Help]
### Summary

Hello, I just installed Storybook and trying out. However, I have stumbled on this issue
![image](https://github.com/storybookjs/storybook/assets/5789670/452dd007-56a6-479f-8abb-f4edf1fa29d8)

The Show Code button supposed to show an example code of the component, but it doesn't get rendered as a developer expects.

If I change the code to export a component directly, the output looks good however, this results in losing control over args in the control menu
![image](https://github.com/storybookjs/storybook/assets/5789670/1e868b51-09ad-46e3-995e-13a76a749b35)
![image](https://github.com/storybookjs/storybook/assets/5789670/0a3ddc03-2fa1-498b-9479-c7efc83ed564)

Is there a way to make the output more DX friendly?

**Comment by Marklb** [comment]
I know very little about SolidJS, but from the repo it seems that a more clear source display is a pending feature. https://github.com/storybookjs/solidjs?tab=readme-ov-file#notes-about-pending-features-
